Bump @docusaurus/preset-classic from 2.1.0 to 2.2.0 (#19)
flipt-io/docs: authentication/methods.mdx

  configure the various authentication methods.
</Note>

#### Token

<Tip>
  When `token` method is enabled and no tokens exist in the backing store, an
  initial token is created and logged in Flipt's server output.
</Tip>

The `token` authentication method supports statically creating authentication tokens.

Once enabled, the `/auth/v1/method/token` API prefix is mounted to Flipt's API.
This section of the API supports the creation of static tokens.

**Example: Token Creation**

The following `curl` command creates a static token with no expiration.
Given authentication is set to `required` then a prior client token will be required to perform this action.

```console
curl -X POST localhost:8080/auth/v1/method/token \
    -H 'Authorization: Bearer gt6P_zIqTnCngfHDCpWb48ob5EBt3PqunUhpofNCNnc=' \
    -H 'Content-Type: application/json' \
    --data '{"name":"access_all_areas","description":"keys to the castle"}'
```

#### OIDC

<Note>
Flipts own UI is designed to support this authentication method natively.
Meaning, once enabled, the UI will support login and present each provider as a login button.

The rest of this information is mostly academic.

> It is only useful if you want to build your own browser application using cookie authentication.

Reviewer: ```suggestion
It is mainly useful if you want to build your own browser application using cookie authentication or understand Flipt's OIDC flow at a lower level.
```

Maybe?

Head over to [OIDC Configuration](/configuration/authentication#method-oidc) documentation to learn how to configure your provider(s).
</Note>

The `OIDC` authentication method is primarily designed to support browser based authentication.
However, it can be manually invoked if such a usecase presents itself.

Once enabled, the `/auth/v1/method/oidc` API prefix is mounted to Flipt's API.
This section of the API supports a generic OAuth 2.0 with OIDC flow.

Flipt's configuration can be defined with multiple simultaneous OIDC providers.
An operator of Flipt chooses a name for each provider and then configures the relevant secrets necessary to authenticate with an OIDC client.

> There are many OIDC providers out there. For example, we have test Flipt with:

Reviewer: ```suggestion
There are many OIDC providers out there. For example, we have tested Flipt with:
```

Reviewer 2: Nice. Is see, this is in the PR you have on this PR.
I will leave this, so you can merge that in easily.

- Google
- Auth0
- Gitlab
- Dex

Each provider has their own way of establishing clients and acquiring the relevant credentials.
You can find futher documentation on leveraging providers like these in our [OIDC Configuration](/configuration/authentication#method-oidc) documentation.

For illustration purposes, let us say we have configured a single provider with `Dex` and named it `dex` (lowercase) in our provider configuration.

This will lead to the following endpoints being available on Flipt:

- `GET /auth/v1/method/oidc/dex/authorize`
- `GET /auth/v1/method/oidc/dex/callback`

> These two endpoints are necessary to support the different legs of the OAuth flow.

Reviewer: ```suggestion
These two endpoints are necessary to support the different legs of the OAuth/OIDC flow.
```

> The first can be requested to obtain an authorize URL directed at the configured instance of Dex.

Reviewer: ```suggestion
The first can be requested to obtain an authorization URL directed at the configured instance of Dex.
```

> The latter is the destination that Dex will redirect client back to.

Reviewer: ```suggestion
The latter is the destination that Dex will redirect the client back to.
```
